packages/web/src/js/FileUploader.ts

```diff
@@ -0,0 +1,192 @@
+    this.isDraggable = 'draggable' in document.createElement('span');
+    this.isMultiple = this.inputElement.multiple;
+    this.isDragging = false;
+    this.fileQueue = [];
+
+    this.init();
+    this.getFileQueue.bind(this);
+    this.clearFileQueue.bind(this);
+  }
+
+  static get NAME() {
+    return NAME;
+  }
+
+  public getFileQueue() {
+    return this.fileQueue;
+  }
+
+  public clearFileQueue() {
+    this.fileQueue = [];
+    this.listElement!.textContent = '';
+  }
+
+  appendToList(file: File) {
+    if (!this.listElement) {
+      if (process.env.NODE_ENV === 'development') {
+        console.warn('Unfortunately, there is no list element created for inserting files to upload ⚠️');
+      }
+
+      return;
+    }
+
+    if (!this.isMultiple) {
+      this.clearFileQueue();
+    }
+
+    // Creating an element to insert into a list
+    const id = file.name.replace(/\.[^/.]+$/, '');
+    const listElementNode = document.createElement('li');
+    listElementNode.setAttribute('id', id);
+    listElementNode.classList.add('FileUploaderAttachment');
+    listElementNode.innerHTML = `
+      <svg width="24" height="24" aria-hidden="true">
+        <use xlink:href="/icons/svg/sprite.svg#file" />
+      </svg>
+      <span class="text-truncate">${file.name}</span>
+      <button type="button" class="FileUploaderAttachment__remove" data-dismiss="${id}">
+        <span class="accessibility-hidden">Remove</span>
+        <svg width="24" height="24" aria-hidden="true">
+          <use xlink:href="/icons/svg/sprite.svg#close" />
+        </svg>
+      </button>
+    `;
+
+    // Appending node into list
+    this.listElement.appendChild(listElementNode);
+
+    // Appending an item into queue
+    this.fileQueue?.push(file);
+
+    // Adding an event for the delete item button
+    const currentItem = SelectorEngine.findOne(`[data-dismiss="${id}"]`);
+    currentItem?.addEventListener('click', (event) => {
+      event.preventDefault();
+      const target = (event.target as HTMLElement)?.dataset?.dismiss;
+      if (target) {
+        this.removeFromQueue(target);
+      }
+    });
+
+    // Reset drag state
+    this.isDragging = false;
+    this.wrapper?.classList.remove(IS_DRAGGING_CLASS_NAME);
+  }
+
+  addToQueue(file: File) {
+    const allowedFileSize = file.size <= FILE_SIZE_LIMIT;
+    const duplicity = this.fileQueue.find((item) => item.name === file.name);
+    const run = this.fileQueue.length === 0 ? true : this.fileQueue.length >= 1 && !duplicity;
+
+    if (run && allowedFileSize) {
+      this.fileQueue?.push(file);
+      this.appendToList(file);
+    } else if (!allowedFileSize) {
+      console.warn('The file size limit has been exceeded ⚠️');
+    } else {
+      console.warn('This file already exists in the queue ⚠️');
+    }
+  }
+
+  removeFromQueue(name: string) {
+    const index = this.fileQueue.findIndex((item) => item.name.replace(/\.[^/.]+$/, '') === name);
```

Comment:
Maybe it would be better to use the `Map` (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) instead of `array` to handle items of the queue as there are baked-in methods for handling additions and removal.

Reply:
ok, done